### PR TITLE
DCOS-20309 Empty state for catalog

### PR DESCRIPTION
**Pull request by Fabs**
To test it remove all the package repositories of a cluster and go to the Catalog (save your universe url). Add packages back and go look the Catalog again.

Also, anyone know how to make the snapshot smaller, It would be by reducing the number of packages for that particular test, there are about 100 packages and 3500 lines of snapshot :/.

closes DCOS-20309.

If you replace those two components on the PackageTab render method, you can get your snapshot.

**To the finish line**
- [x] Fix the snapshot test
- [x] Refactor and improve code style of the solution
- [x] Squash and fix meaningful commit messages

### CODE REVIEW

File: src/js/pages/catalog/PackagesTab.js [hunk 1]
```diff
@@ -38,6 +39,30 @@ const PackagesBreadcrumbs = () => {
...
+const PackagesEmptyState = () => {
+  return (
+    <AlertPanel>
+      <AlertPanelHeader>No package repositories</AlertPanelHeader>
+      <p className="tall">
+        You need at least one package repository with some packages to be
+        able to install packages. For more {" "}
+        <Link
+          target="_blank"
+          to={MetadataStore.buildDocsURI("/administering-clusters/repo")}
```
Comment (ahoskins) [inline src/js/pages/catalog/PackagesTab.js]: Uh oh...this generates: `href="#https://docs.mesosphere.com/1.11/administering-clusters/repo"`

I think just use bare `<a>` tag.

**Comment by juliangieseke** [comment]
Alfred test this

**Comment by juliangieseke** [comment]
```
DEBUG 2018-02-09 21:54:54 proc[cluster.open:teardown]: E: [2018-02-09 21:54:54,655|dcos_launch.platforms.aws|INFO]: Deleting stack: <peep>
DEBUG 2018-02-09 21:54:54 proc[cluster.open:teardown]: E: [2018-02-09 21:54:54,856|dcos_launch.platforms.aws|INFO]: Delete successfully initiated for <peep>
DEBUG 2018-02-09 21:54:54 proc[cluster.open:teardown]: E: [2018-02-09 21:54:54,960|dcos_launch.platforms.aws|INFO]: Waiting for stack operation to complete
DEBUG 2018-02-09 21:54:55 proc[cluster.open:teardown]: E: [2018-02-09 21:54:55,028|dcos_launch.platforms.aws|INFO]: Continuing to wait...
DEBUG 2018-02-09 21:55:55 proc[cluster.open:teardown]: E: [2018-02-09 21:55:55,125|dcos_launch.platforms.aws|INFO]: Continuing to wait...
DEBUG 2018-02-09 21:56:55 proc[cluster.open:teardown]: E: [2018-02-09 21:56:55,283|dcos_launch.platforms.aws|INFO]: Continuing to wait...
DEBUG 2018-02-09 21:57:55 proc[cluster.open:teardown]: E: [2018-02-09 21:57:55,438|dcos_launch.platforms.aws|INFO]: Deleting KeyPair: dcos-ui-system-test-1518212332
DEBUG 2018-02-09 21:57:55 proc.pool: Process exited with code = 0
DEBUG 2018-02-09 21:57:55 cluster.script: Stopping proxy daemon
DEBUG 2018-02-09 21:57:55 proc.pool: Stopping daemon process #1
DEBUG 2018-02-09 21:57:55 proc.pool: Killing daemon process #1
ERROR 2018-02-09 21:57:55 system: Unknown exception while cleaning-up the environment: [Errno 3] No such process
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/dcos_system_test_driver-0.1-py3.6.egg/systemtestdriver/cli/entrypoints.py", line 253, in dcos_system_test_driver
    cluster.teardown()
  File "/usr/local/lib/python3.6/site-packages/dcos_system_test_driver-0.1-py3.6.egg/systemtestdriver/cluster/interfaces/base.py", line 94, in teardown
    daemon_stop(self.proxyDaemon)
  File "/usr/local/lib/python3.6/site-packages/dcos_system_test_driver-0.1-py3.6.egg/systemtestdriver/utils/proc.py", line 324, in daemon_stop
    return DefaultProcessPool.daemon_stop(*args, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/dcos_system_test_driver-0.1-py3.6.egg/systemtestdriver/utils/proc.py", line 266, in daemon_stop
    os.killpg(os.getpgid(threadRec['proc'].pid), signal.SIGTERM)
ProcessLookupError: [Errno 3] No such process
Taking single-use slave mesos-jenkins-52c2c065a01d44c5ba27d1e097695c2a-dcos-ui offline.
```
🤔 @nLight is this the issue you referred to when you said "sometimes dcos-launch isnt able to tear down the cluster"?

**Comment by juliangieseke** [comment]
run system tests

File: src/js/pages/catalog/PackagesTab.js [hunk 2]
```diff
@@ -225,31 +250,36 @@ class PackagesTab extends mixin(StoreMixin) {
...
-          <div className="pod flush-horizontal flush-top">
-            <FieldAutofocus>
-              <FilterInputText
-                className="flex-grow"
-                placeholder="Search catalog"
-                searchString={state.searchString}
-                handleFilterChange={this.handleSearchStringChange}
-              />
-            </FieldAutofocus>
+      if (packages.list == null || packages.list.length === 0) {
```
Comment (juliangieseke) [inline src/js/pages/catalog/PackagesTab.js]: 🤨- not sure if we are supposed to check on the `.list` property directly, I think it should be something like `packages.getItems().length`

**Comment by nLight** [comment]
@juliangieseke I think this is something else. Here the driver failed to stop the proxy server and crashed. Ideally we should try-catch these so they don't crash the build

**Comment by nLight** [comment]
Alfred test this

**Comment by juliangieseke** [comment]
Ah, you're right - cluster is torn down, then it tries to kill the proxy - which seems to be dead already.
=> yes, try-catch would be good

**Comment by nLight** [comment]
https://github.com/mesosphere/dcos-system-test-driver/issues/17

File: src/js/pages/catalog/PackagesTab.js (hunk 2 above)
Comment (juliangieseke) [reply src/js/pages/catalog/PackagesTab.js]: AFAIK someone is responsible to make all this a lot better right now #nopressure 🤪

Comment (Fabs) [reply src/js/pages/catalog/PackagesTab.js]: 🎩 💥 🐇
Not better, insanely great! 

Although we all are responsible hehe, someone is just `facilitating` 😝

**Comment by Fabs** [comment]
run system tests

**Comment by Fabs** [comment]
@juliangieseke and @ahoskins. Addressed your comments. Thanks for the review.

**Comment by nLight** [comment]
Please rebase

**Pull request merged**